### DISCUSSION TITLE
Can I get theme color

### DISCUSSION THREAD

**Discussion by brian8219** [SwiftGen/SwiftGen#993] [Q&A] [answered]
For example I have assets like this
|-StyleA
|        |-PrimaryColor
|-StyleB
        |-PrimaryColor

Is it possible to get color like  Asset.PrimaryColor.color?
And the color is depend on a property that store the current style.

**Comment by djbe** [comment, marked as answer]
SwiftGen won't do this for you (won't generate the code that does it).

What you can do is, assuming both "styles" have the same set of colors, create a protocol such as:
```swift
protocol Style {
  static var primaryColor: ColorAsset { get }
  …
}
```

And conform both `StyleA` and `StyleB` to your protocol like this:
```swift
extension Asset.StyleA: Style {}
extension Asset.StyleB: Style {}
```

Once all of that is set up, you'll probably have some style management service that will check some `UserDefaults` and decide which style to return:
```swift
extension StyleService {
  var activeStyle: Style {
    if someCondition {
      return Asset.StyleA
    } else {
      return Asset.StyleB
    }
  }
```

Something like that? Note that this isn't actually tested code, just something to push you into the right direction. All of the above should be possible *without* a custom template, but if you really want you could create your own template that does this 🤷
